benchmarks/src/main/scala/cats/effect/benchmarks/ParallelBenchmark.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020-2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect.benchmarks
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.implicits.{catsSyntaxParallelTraverse1, toTraverseOps}
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * To do comparative benchmarks between versions:
+ *
+ *     benchmarks/run-benchmark ParallelBenchmark
+ *
+ * This will generate results in `benchmarks/results`.
+ *
+ * Or to run the benchmark from within sbt:
+ *
+ *     jmh:run -i 10 -wi 10 -f 2 -t 4 cats.effect.benchmarks.ParallelBenchmark
+ *
+ * Which means "10 iterations", "10 warm-up iterations", "2 forks", "4 thread".
+ * Please note that benchmarks should be usually executed at least in
+ * 10 iterations (as a rule of thumb), but more is better.
+ */
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class ParallelBenchmark {
+
+  @Param(Array("10000"))
+  var size: Int = _
+
+  @Benchmark
+  def parTraverseCpuTokens100(): Unit =
+    1.to(size).toList.parTraverse(_ => IO(Blackhole.consumeCPU(100L))).void.unsafeRunSync()
+
+  @Benchmark
+  def parTraverseCpuTokens1000(): Unit =
+    1.to(size).toList.parTraverse(_ => IO(Blackhole.consumeCPU(1000L))).void.unsafeRunSync()
```

Comment:
Ah! This isn't quite what I had in mind. The consumption time itself is less interesting to me than the size. We should tune the consumption time so that the `traverse` benchmarks step reasonably, so we're not drowning in noise.

```suggestion
  def parTraverseCpuTokens(): Unit =
    1.to(size).toList.parTraverse(_ => IO(Blackhole.consumeCPU(100L))).void.unsafeRunSync()
```

The `size` parameter can be specified in the `jmh` invocation, which is why `@Param` is helpful.

Reply:
Actually I retract my comment. This is fine :-)